task4/index.js

```diff
@@ -0,0 +1,52 @@
+const ul = document.getElementById("js-ul");
+      
+  const fragment = document.createDocumentFragment();
+  const obj = [
+    {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, 
+    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+  ]
+
+//その1
+Object.values(obj).forEach((value)=>{
+    const li = document.createElement('li');
+    const a = document.createElement('a');
+    const img = document.createElement('img');
+
+    a.textContent = value.text;
+    a.href = value.to;
+    img.src = value.img;
+    img.alt = value.alt;
+
+    fragment.appendChild(li).appendChild(a).insertBefore(img, a.firstChild);
+})
+ul.appendChild(fragment);
```

Comment:
オブジェクトに対してforEachしているのではなく、配列に対してforEachしているので`Object.values()`は不要です。
forRachは配列の要素を繰り返し処理する場合などに使用するメソッドなのでこの問題に関してはこれが最適化と思います。
```suggestion
obj.forEach((value)=>{
    const li = document.createElement('li');
    const a = document.createElement('a');
    const img = document.createElement('img');

    a.textContent = value.text;
    a.href = value.to;
    img.src = value.img;
    img.alt = value.alt;

    fragment.appendChild(li).appendChild(a).insertBefore(img, a.firstChild);
})
ul.appendChild(fragment);
```

Reply:
forRachは配列の要素を繰り返し処理する場合などに使用するメソッドなのでこの問題に関してはこれが最適かと思います。

Reply:
配列とオブジェクトの棲み分けが自分の中で出来ていませんでした。
ご指摘ありがとうございました。
もっと配列について勉強します。